### PR TITLE
feat(Mee6): new site support

### PR DESCRIPTION
**Pull request by DarkVillager**
## Description 

Added support for the new site.
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/M/MEE6/presence.ts [hunk 1]
```diff
@@ -2,59 +2,208 @@ const presence = new Presence({
...
+	Search = "https://i.imgur.com/oGQtnIY.png",
+}
+let strings: Awaited<ReturnType<typeof getStrings>>,
+	oldLang: string = null;
 presence.on("UpdateData", async () => {
 	const presenceData: PresenceData = {
-		largeImageKey: "logo",
-		startTimestamp: browsingTimestamp,
-	};
+			largeImageKey: "logo",
```
Comment (theusaf) [inline websites/M/MEE6/presence.ts:32]:
```suggestion
			largeImageKey: Assets.Logo,
```

File: websites/M/MEE6/presence.ts [hunk 2]
```diff
@@ -2,59 +2,208 @@ const presence = new Presence({
...
+					}
+					case "tutorials": {
+						switch (pathnameSplit[3]) {
+							// eslint-disable-next-line no-undefined
+							case undefined: {
+								presenceData.details = "Viewing all tutorials";
+								break;
+							}
+							default: {
+								presenceData.details = "Reading tutorials about:";
```
Comment (theusaf) [inline websites/M/MEE6/presence.ts:160]: Feels like this could be replaced by an if/else rather than a switch with disabling eslint

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/M/MEE6/presence.ts [hunk 3]
```diff
@@ -2,59 +2,208 @@ const presence = new Presence({
...
-		else {
-			presenceData.details = "Helpdesk viewing:";
-			presenceData.state = title.textContent.replace(" | MEE6 Helpdesk", "");
-			presenceData.smallImageKey = "reading";
+			if (search?.value) {
+				presenceData.details = strings.search;
+				presenceData.state = search.value;
+				presenceData.smallImageKey = Assets.Search;
+			} else if (document.querySelector('[class="article__title"]')) {
+				presenceData.details = "Reading support article about:";
```
Comment (theusaf) [inline websites/M/MEE6/presence.ts:186]: Remove `:` from strings
```suggestion
				presenceData.details = "Reading support article about";
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/M/MEE6/presence.ts [hunk 4]
```diff
@@ -2,59 +2,208 @@ const presence = new Presence({
...
+	const stringTrimmed = string.trim();
+	return (
+		stringTrimmed.charAt(0).toUpperCase() + stringTrimmed.slice(1).toLowerCase()
+	);
+}
+enum Assets {
+	Loading = "https://i.imgur.com/uh6vSQm.gif",
+	Logo = "https://i.imgur.com/AZVCG6a.png",
+	Reading = "https://i.imgur.com/8vMPNni.png",
+	Search = "https://i.imgur.com/oGQtnIY.png",
```
Comment (Bas950) [inline websites/M/MEE6/presence.ts:26]: Please make sure the assets are 512x512

File: websites/M/MEE6/presence.ts (hunk 2 above)
Comment (Bas950) [reply websites/M/MEE6/presence.ts:160]: Yea, if case would be better here, if (!pathnameSplit[3]) {} else {}

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/M/MEE6/presence.ts (hunk 4 above)
Comment (DarkVillager) [reply websites/M/MEE6/presence.ts:26]: At least or are?

Comment (Bas950) [reply websites/M/MEE6/presence.ts:26]: Are

File: websites/M/MEE6/presence.ts [hunk 5]
```diff
@@ -2,59 +2,202 @@ const presence = new Presence({
...
+			browse: "general.browsing",
+			search: "general.searchFor",
+			viewCategory: "general.viewCategory",
+			viewHome: "general.viewHome",
+		},
+		await presence.getSetting<string>("lang").catch(() => "en")
+	);
+}
+async function capitalizeFirstLetter(string: string) {
+	const stringTrimmed = string.trim();
```
Comment (EncryptedDev) [inline websites/M/MEE6/presence.ts:17]: Variable isn't really needed, you may as well just add it to the return

File: websites/M/MEE6/presence.ts [hunk 6]
```diff
@@ -2,59 +2,202 @@ const presence = new Presence({
...
+						presenceData.details = "Viewing custom bot options";
+						break;
+					}
+					case "pro":
+					case "premium": {
+						presenceData.details = "Viewing premium & pro options";
+						break;
+					}
+					case "careers": {
+						presenceData.details = "Viewing open careere positions";
```
Comment (EncryptedDev) [inline websites/M/MEE6/presence.ts:86]:
```suggestion
						presenceData.details = "Viewing open careers";
```

File: websites/M/MEE6/presence.ts [hunk 7]
```diff
@@ -2,59 +2,202 @@ const presence = new Presence({
...
+						break;
+					}
+					case "memberships": {
+						presenceData.details = "Managing their subscriptions";
+						break;
+					}
+					case "dashboard": {
+						switch (pathnameSplit[4]) {
+							case "": {
+								if (pathnameSplit[3] !== "") {
```
Comment (EncryptedDev) [inline websites/M/MEE6/presence.ts:106]: Empty string is falsy 
```suggestion
								if (pathnameSplit[3]) {
```

File: websites/M/MEE6/presence.ts [hunk 8]
```diff
@@ -2,59 +2,202 @@ const presence = new Presence({
...
+							}
+							default: {
+								if (
+									document.querySelector(
+										'[class="uppercase text-dark-400 text-xs font-semibold"]'
+									) &&
+									pathname.includes("commands")
+								) {
+									presenceData.details = "Editing custom command";
+									presenceData.state = await capitalizeFirstLetter(
```
Comment (EncryptedDev) [inline websites/M/MEE6/presence.ts:125]: `await` shouldn't be needed here

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — APPROVED** [review]

File: websites/M/MEE6/presence.ts [hunk 9]
```diff
@@ -2,59 +2,202 @@ const presence = new Presence({
...
+			viewCategory: "general.viewCategory",
+			buttonViewPage: "general.buttonViewPage",
+			viewHome: "general.viewHome",
+		},
+		await presence.getSetting<string>("lang").catch(() => "en")
+	);
+}
+function capitalizeFirstLetter(string: string) {
+	return (
+		string.charAt(0).toUpperCase().trim() + string.slice(1).toLowerCase().trim()
```
Comment (Bas950) [inline websites/M/MEE6/presence.ts:19]: You should probably trim the string first, cause rn you are getting the character at position 0, which would be 1 character and then trimming it, which does nothing...

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/M/MEE6/presence.ts (hunk 9 above)
Comment (DarkVillager) [reply websites/M/MEE6/presence.ts:19]: oh good point

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**